src/MySqlEventStore.php

```diff
@@ -552,11 +512,166 @@ public function stopProjection(string $name): void
         ]);
     }
 
+    public function fetchStreamNames(
+        ?string $filter,
+        bool $regex,
+        ?MetadataMatcher $metadataMatcher,
+        int $limit,
+        int $offset
+    ): array {
+        if (null === $filter && $regex) {
+            throw new Exception\InvalidArgumentException('No regex pattern given');
+        }
+
+        if ($regex && false === @preg_match("/$filter/", '')) {
+            throw new Exception\InvalidArgumentException('Invalid regex pattern given');
+        }
+
+        $where = [];
+        $values = [];
+
+        if (null !== $filter && $regex) {
+            $where[] = '`real_stream_name` REGEXP :filter ';
+            $values[':filter'] = $filter;
+        } elseif (null !== $filter && ! $regex) {
+            $where[] = '`real_stream_name` = :filter ';
+            $values[':filter'] = $filter;
+        }
+
+        list($where, $values) = $this->createWhereClauseForMetadata($metadataMatcher, $where, $values);
+
+        $whereCondition = implode(' AND ', $where);
+        if (! empty($whereCondition)) {
+            $whereCondition = 'WHERE ' . $whereCondition;
+        }
+
+        $query = <<<SQL
+SELECT `real_stream_name` FROM $this->eventStreamsTable
+$whereCondition
+ORDER BY `no`
```

Comment:
We should consider if the index can be used for sorting. I think `ORDER BY real_stream_name` is slower than `ORDER BY no`.